### ISSUE TITLE
Error when running "buck build example" from openspatial-android-sdk/sdk

### ISSUE THREAD

**Issue by minakhan01** [openspatial/openspatial-android-sdk#1]
Using buckd.
[+] PROCESSING BUCK FILES...0.3s
java.util.concurrent.ExecutionException: java.lang.RuntimeException: Invalid version string: build-tools-17.0.0
    at com.google.common.util.concurrent.AbstractFuture$Sync.getValue(AbstractFuture.java:299)
    at com.google.common.util.concurrent.AbstractFuture$Sync.get(AbstractFuture.java:286)
    at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:116)
    at com.facebook.buck.command.Build.executeBuild(Build.java:239)
    at com.facebook.buck.command.Build.executeAndPrintFailuresToConsole(Build.java:270)
    at com.facebook.buck.cli.BuildCommand.runWithoutHelp(BuildCommand.java:368)
    at com.facebook.buck.cli.AbstractCommand.run(AbstractCommand.java:133)
    at com.facebook.buck.cli.BuckCommand.run(BuckCommand.java:80)
    at com.facebook.buck.cli.Main.runMainWithExitCode(Main.java:654)
    at com.facebook.buck.cli.Main.tryRunMainWithExitCode(Main.java:989)
    at com.facebook.buck.cli.Main.runMainThenExit(Main.java:1046)
    at com.facebook.buck.cli.Main.nailMain(Main.java:1075)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at com.martiansoftware.nailgun.NGSession.run(NGSession.java:338)
Caused by: java.lang.RuntimeException: Invalid version string: build-tools-17.0.0
    at com.facebook.buck.util.VersionStringComparator.compare(VersionStringComparator.java:40)
    at com.facebook.buck.android.AndroidPlatformTarget$3.compare(AndroidPlatformTarget.java:347)
    at com.facebook.buck.android.AndroidPlatformTarget$3.compare(AndroidPlatformTarget.java:1)
    at java.util.TimSort.countRunAndMakeAscending(TimSort.java:356)
    at java.util.TimSort.sort(TimSort.java:216)
    at java.util.Arrays.sort(Arrays.java:1512)
    at java.util.ArrayList.sort(ArrayList.java:1454)
    at java.util.Collections.sort(Collections.java:175)
    at com.facebook.buck.android.AndroidPlatformTarget.pickNewestBuildToolsDir(AndroidPlatformTarget.java:342)
    at com.facebook.buck.android.AndroidPlatformTarget.createFromDefaultDirectoryStructure(AndroidPlatformTarget.java:276)
    at com.facebook.buck.android.AndroidPlatformTarget$AndroidWithGoogleApisFactory.newInstance(AndroidPlatformTarget.java:421)
    at com.facebook.buck.android.AndroidPlatformTarget.getTargetForId(AndroidPlatformTarget.java:196)
    at com.facebook.buck.cli.Main$1.get(Main.java:751)
    at com.facebook.buck.cli.Main$1.get(Main.java:1)
    at com.facebook.buck.step.ExecutionContext.getAndroidPlatformTarget(ExecutionContext.java:170)
    at com.facebook.buck.android.DxStep.getShellCommandInternal(DxStep.java:117)
    at com.facebook.buck.shell.ShellStep.getShellCommand(ShellStep.java:215)
    at com.facebook.buck.shell.ShellStep.getDescription(ShellStep.java:246)
    at com.facebook.buck.step.DefaultStepRunner.runStepForBuildTarget(DefaultStepRunner.java:56)
    at com.facebook.buck.rules.CachingBuildEngine.executeCommandsNowThatDepsAreBuilt(CachingBuildEngine.java:702)
    at com.facebook.buck.rules.CachingBuildEngine.access$4(CachingBuildEngine.java:678)
    at com.facebook.buck.rules.CachingBuildEngine$1.apply(CachingBuildEngine.java:269)
    at com.facebook.buck.rules.CachingBuildEngine$1.apply(CachingBuildEngine.java:1)
    at com.google.common.util.concurrent.Futures$ChainingListenableFuture.run(Futures.java:906)
    at com.google.common.util.concurrent.Futures$1$1.run(Futures.java:635)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
BUILD FAILED: Unexpected internal error (if you are using buckd, you should restart it).

I have tried restarting buck and buckd, and I also have android build tools 17.
Could you please help me with this issue?
Thanks!

**Comment by minakhan01** [comment]
$ export ANDROID_SDK=path_to_android_sdk
  $ export PATH=$PATH:$ANDROID_SDK/sdk/platform-tools
  $ export OPENSPATIAL_SDK=path_to_openspatial_sdk

Which of the following is correct? I have tried both and both times I get "Invalid version string: build-tools-17.0.0" error. I know that I have build-tools-17.0.0 folder in /Users/khan32m/Library/Android/sdk/build-tools

OPTION A:
 $ export ANDROID_SDK=/Users/khan32m/Library/Android
 $ export PATH=$PATH:$ANDROID_SDK/sdk/platform-tools
 $ export OPENSPATIAL_SDK=/Users/khan32m/git_repos/openspatial-android-sdk

OPTION B:
 $ export ANDROID_SDK=/Users/khan32m/Library/Android
 $ export PATH=$PATH:$ANDROID_SDK/sdk/platform-tools
 $ export OPENSPATIAL_SDK=/Users/khan32m/git_repos/openspatial-android-sdk

**Comment by ndahlquist** [comment]
@minakhan01, I've filed an issue here: https://github.com/facebook/buck/issues/425

tldr is that buck doesn't like the directory name format `build-tools-xx.x.x`. As a workaround, you could rename these to `xx.x.x`.
